src/formatter/formatter.vala

```diff
@@ -0,0 +1,278 @@
+using Gee;
+
+class Vls.Formatter : Object {
+    private Pair<Vala.SourceFile, Compilation> input;
+    private Lsp.FormattingOptions options;
+
+    public Formatter (Lsp.FormattingOptions options, Pair<Vala.SourceFile, Compilation> input) {
+        this.options = options;
+        this.input = input;
+    }
+
+    public string? format (out Lsp.TextEdit edit, out Jsonrpc.ClientError error) {
+        error = 0;
+        edit = null;
+        File cfg;
+        try {
+            cfg = this.generate_uncrustify_config ();
+        } catch (Error e) {
+            return "Error creating config: %s".printf (e.message);
+        }
+        var original = File.new_for_path (this.input.first.filename);
```

Comment:
`uncrustify` can read from stdin and write to stdout. So we should be doing that instead of reading temporary files.

Reply:
It seems that "SEARCH_PATH_FROM_ENVP", that I think I require is not available even in newer distros (Like Fedora 35). This means again that uncrustify has to be installed in either /bin, /usr/bin and the standard windows paths.

Reply:
How should I continue here? Just say "I don't care that uncrustify is not installed on /bin or /usr/bin" or do e.g.

```
try {
  // Try with "uncrustify"
} catch (...) {
  try {
    // Try with /usr/local/bin/uncrustify
  }
}
```
We could do the latter and as soon as `SEARCH_PATH_FROM_ENVP` is available in e.g. Debian Stable/Testing switch to using it.

Reply:
Ah, that's right. I may have removed `SubprocessLauncher` by mistake then, so we could use that with `SubprocessLauncher.set_environ()` and `Environ.get()` instead, and make a note of it in the code.